src/test/java/cohen/mvc/FreebiePostsControllerTest.java

```diff
@@ -0,0 +1,40 @@
+package cohen.mvc;
+
+import cohen.FreebieService;
+import cohen.json.PostListInfo;
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.plugins.RxJavaPlugins;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import org.junit.jupiter.api.Test;
+
+import javax.swing.*;
+
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.verify;
+
+public class FreebiePostsControllerTest
+{
+    static
+    {
+        RxJavaPlugins.setIoSchedulerHandler(scheduler -> Schedulers.trampoline());
+        RxJavaPlugins.setNewThreadSchedulerHandler(scheduler -> Schedulers.trampoline());
+    }
+
+    @Test
+    void refreshPosts()
+    {
+        //given
+        FreebieService service = mock();
+        JTextArea postList = mock();
+        FreebiePostsController controller = new FreebiePostsController(service, postList);
+        PostListInfo postListInfo = mock();
+        Observable<PostListInfo> observableList = Observable.just(postListInfo);
+        doReturn(observableList).when(service).getPostList();
+
+        //when
+        controller.refreshPosts();
+
+        //then
+        verify(service).getPostList();
```

Comment:
You need to test that what happens in the subscribe() part happens. 
It's not enough to just make sure getPostList() is called.